src/main/java/net/snowflake/client/config/SFConnectionConfigParser.java

```diff
@@ -127,10 +124,23 @@ public static ConnectionParameters buildConnectionParameters() throws SnowflakeS
     }
   }
 
-  private static String createUrl(String account, Map<String, String> fileConnectionConfiguration) {
-    String host = String.format("%s.snowflakecomputing.com", account);
+  private static String createUrl(Map<String, String> fileConnectionConfiguration)
+      throws SnowflakeSQLException {
+    Optional<String> maybeAccount = Optional.ofNullable(fileConnectionConfiguration.get("account"));
+    Optional<String> maybeHost = Optional.ofNullable(fileConnectionConfiguration.get("host"));
+    String host =
+        maybeHost.orElse(
+            maybeAccount
+                .map(acnt -> String.format("%s.snowflakecomputing.com", acnt))
+                .orElse(null));
+    if (host == null || host.isEmpty()) {
+      logger.warn("Neither host nor account is specified in connection parameters");
+      throw new SnowflakeSQLException(
+          "Unable to connect because neither host nor account is specified in connection parameters");
+    }
     String port = fileConnectionConfiguration.get("port");
     String protocol = fileConnectionConfiguration.get("protocol");
+    logger.debug("Host created using parameters from connection configuration file: {}", host);
```

Comment:
This should be logged right after host is calculated? Preferably before the exception?

Reply:
ok